### DISCUSSION TITLE
Genie commit option not available for commit messages generation by Genie in Source control for github repository

### DISCUSSION THREAD

**Discussion by sebastianneu** [posit-dev/positron#5393] [Q&A] [answered]
Dear all,

is anyone using Genie as well? In general, Genie is working for me in the standard way. This means, one can use it in the editor or as Genie directly. In VScode, when working within git repositories, one can use "Genie commit" option to generate commit messages before committing and pushing changes. This is not available for Positron.
Does anyone experience the same?

Thanks and kind regards

**Comment by juliasilge** [comment]
Is this a VS Code extension? My initial searching did not identify for me what you might be talking about. Can you share more details and perhaps a screenshot of what is not working for you?

**Comment by sebastianneu** [comment]
This is a VS Code extension. It is available for VS Code and Positron. It is based on the same github repository. It is also working the same for VS Code and Positron, but it seems there is not the same option to use "Genie: Generate a commit message". The following two pictures show first the one from VS Code and second from Positron where there is not such a button. Also behind the three dots nothing is available.

![grafik](https://github.com/user-attachments/assets/9a2d75bd-54ed-4b22-a220-a397630da472)

![grafik](https://github.com/user-attachments/assets/29a9a3a3-39a9-4462-8828-9e44dae65d38)

**Reply by juliasilge** [reply]
Can you share specifically what extension you are using? The only one called just plain "Genie" doesn't look like what you're talking about:

![Screenshot 2024-11-19 at 8 42 13 AM](https://github.com/user-attachments/assets/189cfbf2-d2d8-4ab4-a410-946731222854)

A link to the extension on GitHub or the extension marketplace would be really helpful.

**Comment by sebastianneu** [comment]
Sorry for not being more specific. For VS Code as well as for Positron it is "ChatGPT - Genie AI" - both have the repository "https://github.com/ai-genie/chatgpt-vscode.git".

**Reply by juliasilge** [reply]
Thank you! I notice that there is no actual code in that repo, so there's not much for me to take a look at, to see what might be going on. There [are some issues about the commit message functionality](https://github.com/search?q=repo%3Aai-genie%2Fchatgpt-vscode+Generate+a+commit+message&type=issues), so you might check those out.

Positron has not made any changes to the Git or Chat functionality, so I would be surprised if this problem is due to something Positron is doing. However, if you have the bandwidth to dig a bit deeper, I would recommend that you try out the extension in a non-Positron, non-VSCode editor, such as VSCodium. If it works in VSCodium, we would be pretty interested in hearing more about your problem, because then it _may_ be something Positron is doing. If it doesn't work in VSCodium, then it is likely the extension itself not being set up to work with IDEs other than VS Code.

**Reply by sebastianneu** [reply]
Thanks for the hint. I tried it with VSCodium and it is the same as with Positron. This finally means, that some of the functionality of ChatGPT - Genie is working pretty the same as in VS Code - like explainign code or optimize, etc, But some not - like "Genie commit", which is a pitty as it is very helpful.

**Reply by juliasilge** [reply, marked as answer]
Thanks for checking! I would recommend that you [open an issue on their repo](https://github.com/ai-genie/chatgpt-vscode/issues) explaining that this particular feature does not work in IDEs that are otherwise `.vsix` compatible like VSCodium and Positron.
